daemon/container_operations.go

```diff
@@ -354,6 +354,14 @@ func (daemon *Daemon) findAndAttachNetwork(container *container.Container, idOrN
 		if container.Managed || !n.Info().Dynamic() {
 			return n, nil, nil
 		}
+		// Throw an error if the container is already attached to the network
+		if container.NetworkSettings.Networks != nil {
+			networkName := n.Name()
+			containerName := strings.TrimPrefix(container.Name, "/")
+			if network, ok := container.NetworkSettings.Networks[networkName]; ok && network.EndpointID != "" {
+				return n, nil, types.ForbiddenErrorf("%s is already attached to network %s", containerName, networkName)
```

Comment:
If we do not want to handle this silently (note that if we do, we should still check if the provided custom IP-addresses (if any)) are the same as the existing ones, otherwise produce an error), I think this should be a `types.BadRequestErrorf`, not a "forbidden" (HTTP 403)

Actually, a `Conflict` (HTTP 409) might be more appropriate https://github.com/moby/moby/blob/c19b9b1885ec6c9a9fa91211111691f994aef5b7/errdefs/defs.go#L13-L17


Also, instead of handling this situation here, could we delegate this it to `daemon.clusterProvider.AttachNetwork` ? It feels like this is something that that function should handle https://github.com/moby/moby/blob/d912e9460e287238f879de791595193543ab2202/daemon/cluster/networks.go#L181-L182

Reply:
thanks for the suggestion, I'd rather return an rrror and not silently return, a `409` seems more apt